test/y.bls.test.ts

```diff
@@ -94,6 +100,39 @@ describe('bls account', function () {
     expect(ret).to.equal('0x')
   })
 
+  it('aggregated sig validation must succeed if off-chain UserOp sig succeeds', async () => {
+    // regression AA-119: prevent off-chain signature success and on-chain revert.
+    // "broken account" uses different public-key during construction and runtime.
+    const brokenAccountFactory = await new BrokenBLSAccountFactory__factory(etherSigner).deploy(entrypoint.address, blsAgg.address)
+    // const brokenAccountFactory = await new BLSAccountFactory__factory(etherSigner).deploy(entrypoint.address, blsAgg.address)
+    const deployTx = await brokenAccountFactory.populateTransaction.createAccount(0, signer1.pubkey)
+    const res = await brokenAccountFactory.provider.call(deployTx)
+    const acc = brokenAccountFactory.interface.decodeFunctionResult('createAccount', res)[0]
+    await fund(acc)
+    const userOp = await fillUserOp({
+      sender: acc,
+      initCode: hexConcat([brokenAccountFactory.address, deployTx.data!])
+    }, entrypoint)
+    const requestHash = await blsAgg.getUserOpHash(userOp)
+    const signature = userOp.signature = hexConcat(signer1.sign(requestHash))
+
+    // and sig validation should fail:
+    const singleOpSigCheck = await blsAgg.validateUserOpSignature(userOp).then(() => 'ok', e => e.message) as string
+
+    // above account should fail on-chain:
+    const beneficiary = createAddress()
+    const handleRet = await entrypoint.callStatic.handleAggregatedOps([
+      {
+        userOps: [userOp],
+        aggregator: blsAgg.address,
+        signature
+      }
+    ], beneficiary).then(() => 'ok', e => e.errorName) as string
+
+    expect(`${singleOpSigCheck},${handleRet}`)
```

Comment:
I am having a problem understanding what is being asserted here. Can we have a more explicit `expect`?

Reply:
I want to assert that both are "ok", and still in the log see the failures of each.
this is because the first commit adds the tests and fails (the simulation succeeds but the handleOps fails), and the 2nd commit fixes the issues.